crates/trie/sparse/src/trie.rs

```diff
@@ -698,11 +748,36 @@ impl RevealedSparseTrie {
                         .branch_value_stack_buf
                         .resize(buffers.branch_child_buf.len(), Default::default());
                     let mut added_children = false;
+
+                    let mut tree_mask_values = Vec::new();
+                    let mut hash_mask_values = Vec::new();
```

Comment:
why are these vecs and not bitmasks (`TrieMask`s) from the start

Reply:
it requires checking the children set in `state_mask` to correctly set the bits in `tree_mask` and `hash_mask`, so I would suggest we do it in a follow-up PR because it requires changes to `alloy-trie`